### ISSUE TITLE
Magic Map issue

### ISSUE THREAD

**Issue by Glumac777** [TeamTwilight/twilightforest#1986]
### NeoForge Version

20.4.147-beta

### Twilight Forest Version

1.20.4-4.4.2147

### Steps to Reproduce

I bring the magic map with myself;
Join the Twilight forest;
Said map only shows icons of structures, but no topography of the area itself.

### What You Expected

Entered the twilight forest with said magic map, in hopes of finding more structures.

### What Happened Instead

The map itself has icons of structures, but just the icons, no topography of the area itself.

### Additional Details

Sometimes the map starts working again after some while (took an hour last time I had this issue).

### Please Read and Confirm The Following

- [X] I have confirmed this bug can be replicated without the use of Optifine.
- [X] I have confirmed the details provided in this report are concise as possible and does not contained vague information (ie. Versions are properly recorded, answers to questions are clear).
- [X] I have confirmed this issue is unique and has not been reported already.

**Comment by Andromander** [comment]
Reproduced, as in I had it happen to me randomly. No reliable way to make it happen.

Client Log: https://gist.github.com/Andromander/59afbe4cf054fc868975e6c5e7578109

Main points: the error happened when I joined the server, meaning it's likely the error happens only when the map is in the player inventory. As for the source: https://github.com/TeamTwilight/twilightforest/blob/1.20.x/src/main/java/twilightforest/network/MagicMapPacket.java#L48. Level is null, so it throws, but because it fails, it never gets a chance to update again.
